File: angularapp/src/app/services/property-inquiry.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PropertyInquiry } from '../models/property-inquiry.model';

@Injectable({
  providedIn: 'root'
})
export class PropertyInquiryService {

  private readonly backendUrl="http://localhost:8080/api/inquiries"
  constructor(private readonly http:HttpClient) { }

  getAllInquiries():Observable<PropertyInquiry[]>{
    return this.http.get<PropertyInquiry[]>(this.backendUrl);
  }

  getInquiriesByUserId(userId:number):Observable<PropertyInquiry[]>{
    return this.http.get<PropertyInquiry[]>(`${this.backendUrl}/user/${userId}`);
  }

  addInquiry(inquiry:PropertyInquiry):Observable<PropertyInquiry>{
    return this.http.post<PropertyInquiry>(this.backendUrl,inquiry);
  }

  updateInquiry(inquiryId:number,inquiry:Partial<PropertyInquiry>):Observable<PropertyInquiry>{
    return this.http.put<PropertyInquiry>(`${this.backendUrl}/${inquiryId}`,inquiry);
  }

  deleteInquiry(inquiryId:number):Observable<void>{
    return this.http.delete<void>(`${this.backendUrl}/${inquiryId}`);
  }

  getAdminControlPanelData() : Observable<Map<string, number>> {
    return this.http.get<Map<string, number>>(`${this.backendUrl}/adminControlPanel`)
  }
}

File: springapp/src/main/java/com/examly/springapp/service/PropertyInquiryService.java
package com.examly.springapp.service;

import java.util.List;
import java.util.Map;
import java.util.Optional;

import com.examly.springapp.model.PropertyInquiry;

/*
 * Service interface for managing Property Inquiry operations.
 */
public interface PropertyInquiryService {

    /*
     * Adds a new Property Inquiry.
     */
    PropertyInquiry addInquiry(PropertyInquiry propertyInquiry);

    /*
     * Retrieves a Property Inquiry by its ID.
     */
    Optional<PropertyInquiry> getInquiryById(long inquiryId);

    /*
     * Retrieves all Property Inquiries.
     */
    List<PropertyInquiry> getAllInquiries();

    /*
     * Updates an existing Property Inquiry.
     */
    PropertyInquiry updateInquiry(long inquiryId, PropertyInquiry propertyInquiry);

    /*
     * Deletes a Property Inquiry by its ID.
     */
    void deleteInquiry(long inquiryId);

    /*
     * Retrieves all Property Inquiries associated with a specific user.
     */
    List<PropertyInquiry> getInquiriesByUserId(long userId);

    Map<String, Integer> getAdminControlPanelData();
}